### PR TITLE
8283606: Tests may fail with zh locale on MacOS

### PR DESCRIPTION
**Pull request by Nagata-Haruhito**
I would like to backport JDK-8283606 to JDK17u. This fix can be applied cleanly.
I tried to run these tests on Japanese Windows 10, and all tests are passed.

But, I am not a author, so need a sponsor.

Thank you.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8283606](https://bugs.openjdk.org/browse/JDK-8283606): Tests may fail with zh locale on MacOS

### Reviewers
 * [Christoph Langer](https://openjdk.org/census#clanger) (@RealCLanger - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk17u-dev pull/963/head:pull/963` \
`$ git checkout pull/963`

Update a local copy of the PR: \
`$ git checkout pull/963` \
`$ git pull https://git.openjdk.org/jdk17u-dev pull/963/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 963`

View PR using the GUI difftool: \
`$ git pr show -t 963`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk17u-dev/pull/963.diff">https://git.openjdk.org/jdk17u-dev/pull/963.diff</a>

</details>

### CODE REVIEW

**Review by RealCLanger — APPROVED** [review]
Looks good. Seems clean apart from some copyright header fixes.

Please enable Github Actions in your fork repo and run the tests there to get some quality gate check. Thanks.

**Comment by RealCLanger** [comment]
Hi @Nagata-Haruhito,
for JDK-Updates you must not just integrate a PR once its reviewed but you also need to seek maintainer approval. Please read https://wiki.openjdk.org/display/JDKUpdates/How+to+contribute+or+backport+a+fix for more details. In short, please add a jdk17u-fix-request label to the JBS bug and add a comment about why you want to backport the fix, what are the involved risks and what testing you did.

**Comment by Nagata-Haruhito** [comment]
Thank you for your advice.
Will I need retype "/integrate" after JBS label and comment is updated ?

**Comment by RealCLanger** [comment]
> Thank you for your advice. Will I need retype "/integrate" after JBS label and comment is updated ?

No, I just approved the backport.

/sponsor

**Pull request closed without merging**
